### PR TITLE
chore: move stdune tests to stdune 

### PR DESCRIPTION
**Pull request by Alizter**

### CODE REVIEW

**Comment by rgrinberg** [comment]
The `pp_tests` should be moved to the pp package actually

**Comment by Alizter** [comment]
@rgrinberg Since that is vendored, its gonna be a pain to have tests for it there. I've moved it back to the expect-tests dir.

**Comment by rgrinberg** [comment]
Okay, that's fine for now. Can you add open a PR to ocaml-dune/pp to add it?

Also, can it be a separate test library in expect-tests/pp? Right now I think you attached it to `dune_unit_tests` which isn't right.

**Comment by Alizter** [comment]
I'll make a PR adding it upstream.

**Comment by Alizter** [comment]
@rgrinberg Looks like upstream already has these tests and more. We need to update the vendor.

**Comment by Alizter** [comment]
@rgrinberg I've added the more comprehensive tests from upstream and updated the vendor for pp. I find it kind of weird that pp is packaged as dyn.pp, what is the reason for this?

**Comment by rgrinberg** [comment]
Because of the flawed way we do vendoring. We have public libraries that depend on `pp`, hence `pp` needs to be public. But we can't use the `pp` package name because we aren't publishing this package. I'll fix this nonsense soon enough.

File: vendor/pp/test/dune [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(library
+ (name dune_pp_tests)
```
Comment (rgrinberg) [inline vendor/pp/test/dune:2]: Is there a corresponding commit for this in the pp repo?

Comment (Alizter) [reply vendor/pp/test/dune:2]: The pp_repo already has these tests. We replace the Dune files ourselves like always.

**Comment by rgrinberg** [comment]
Okay, I see. But the commit message here is confusing. You're doing two things:

* Updating pp and as a consequence removing the duplicate tests
* Moving tests to the stdune directory

These should be done separately. Preferably in the same order mentioned above.

**Comment by Alizter** [comment]
@rgrinberg OK let me split it up.

**Pull request merged**